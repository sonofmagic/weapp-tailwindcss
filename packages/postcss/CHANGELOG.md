# @weapp-tailwindcss/postcss

## 1.0.0-alpha.1

### Major Changes

- [`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: monorepo changes

### Patch Changes

- Updated dependencies [[`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.1

## 0.0.1-alpha.0

### Patch Changes

- Updated dependencies [[`d50d95d`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d50d95d04e1c6b7c3ff32acc0d9894d3c0f06d22)]:
  - @weapp-tailwindcss/shared@0.0.1-alpha.0
