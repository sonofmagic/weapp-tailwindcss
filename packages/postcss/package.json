{
  "name": "@weapp-tailwindcss/postcss",
  "version": "1.0.0-alpha.2",
  "description": "@weapp-tailwindcss/postcss",
  "author": "ice breaker <1324318532@qq.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sonofmagic/monorepo-template.git",
    "directory": "packages/bar"
  },
  "bugs": {
    "url": "https://github.com/sonofmagic/monorepo-template/issues"
  },
  "keywords": [],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    },
    "./types": {
      "types": "./dist/types.d.ts",
      "import": "./dist/types.mjs",
      "require": "./dist/types.js"
    }
  },
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "tsup --watch --sourcemap",
    "build": "tsup",
    "test": "vitest run",
    "test:dev": "vitest",
    "release": "pnpm publish",
    "lint": "eslint .",
    "lint:fix": "eslint . --fix"
  },
  "publishConfig": {
    "exports": {
      ".": {
        "types": "./dist/index.d.ts",
        "import": "./dist/index.mjs",
        "require": "./dist/index.js"
      },
      "./types": {
        "types": "./dist/types.d.ts",
        "import": "./dist/types.mjs",
        "require": "./dist/types.js"
      }
    },
    "main": "./dist/index.js",
    "module": "./dist/index.mjs",
    "types": "./dist/index.d.ts"
  },
  "dependencies": {
    "@csstools/postcss-is-pseudo-class": "^5.0.1",
    "@weapp-core/escape": "~3.0.2",
    "@weapp-tailwindcss/shared": "workspace:*",
    "postcss": "~8.4.49",
    "postcss-rem-to-responsive-pixel": "~6.0.2",
    "postcss-selector-parser": "~7.0.0"
  },
  "devDependencies": {
    "@weapp-tailwindcss/mangle": "workspace:*"
  }
}
