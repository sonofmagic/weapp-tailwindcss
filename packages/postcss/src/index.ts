// 统一导出入口，供外部调用端按需引用核心能力
export * from './handler'
export {
  createStylePipeline,
  type PipelineNodeContext,
  type PipelineNodeCursor,
  type PipelineStage,
  type ResolvedPipelineNode,
  type StyleProcessingPipeline,
} from './pipeline'
export { createInjectPreflight } from './preflight'
export { internalCssSelectorReplacer } from './shared'
export * from './types'
