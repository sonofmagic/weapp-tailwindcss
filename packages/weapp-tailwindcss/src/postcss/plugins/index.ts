import type { AcceptedPlugin } from 'postcss'
// @ts-expect-error
import postcssIsPseudoClass from '@csstools/postcss-is-pseudo-class'
import postcssRem2rpx from 'postcss-rem-to-responsive-pixel'
import type { IStyleHandlerOptions } from '../../types'
import { postcssWeappTailwindcssPrePlugin } from './pre'
import { postcssWeappTailwindcssPostPlugin } from './post'
import { createContext } from './ctx'

export function getPlugins(options: IStyleHandlerOptions) {
  const ctx = createContext()
  options.ctx = ctx
  const plugins: AcceptedPlugin[] = [
    ...(options.postcssOptions?.plugins ?? []),
    postcssWeappTailwindcssPrePlugin(options),
    postcssIsPseudoClass({
      specificityMatchingName: 'weapp-tw-ig',
    }),
  ]
  if (options.rem2rpx) {
    plugins.push(
      postcssRem2rpx(
        typeof options.rem2rpx === 'object'
          ? options.rem2rpx
          : {
              rootValue: 32,
              propList: ['*'],
              transformUnit: 'rpx',
            },
      ),
    )
  }
  plugins.push(postcssWeappTailwindcssPostPlugin(options))
  return plugins
}

export { postcssWeappTailwindcssPostPlugin } from './post'
export { postcssWeappTailwindcssPrePlugin } from './pre'
// @ts-expect-error
export { default as postcssIsPseudoClass } from '@csstools/postcss-is-pseudo-class'
export { default as postcssRem2rpx } from 'postcss-rem-to-responsive-pixel'
