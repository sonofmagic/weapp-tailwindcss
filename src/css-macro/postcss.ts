import type { PluginCreator } from 'postcss'
import { matchCustomPropertyFromValue, ifdef, ifndef } from './constants'
export interface Options {}

const creator: PluginCreator<Options> = () => {
  return {
    postcssPlugin: 'postcss-weapp-tw-css-macro-plugin',
    prepare() {
      return {
        AtRule(atRule, helper) {
          if (atRule.name === 'media') {
            const values: string[] = []
            matchCustomPropertyFromValue(atRule.params, (arr) => {
              values.push(arr[1])
            })
            if (values.length > 0) {
              const isNegative = atRule.params.includes('not')
              const text = values.join(' ')
              const comment = isNegative ? ifndef(text) : ifdef(text)

              atRule.before([
                helper.comment({
                  raws: {
                    before: '\n'
                  },
                  text: comment.start
                }),
                ...atRule.nodes, // .map((x) => x.clone()),
                helper.comment({
                  raws: {
                    before: '\n'
                  },
                  text: comment.end
                })
              ])
              atRule.remove()
            }
          }

          /*  #ifdef  %PLATFORM%  */
          // 平台特有样式
          /*  #endif  */
        },
        CommentExit(comment) {
          if (/#(?:ifn?def|endif)/.test(comment.text)) {
            comment.raws.left = ' '
            comment.raws.right = ' '
          }
        }
      }
    }
  }
}

creator.postcss = true

export default creator
