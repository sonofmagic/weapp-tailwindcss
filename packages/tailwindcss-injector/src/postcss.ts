import type { Input, PluginCreator } from 'postcss'
import type { Config } from 'tailwindcss'
import type { Options } from './types'
import postcss from 'postcss'
import tailwindcss from 'tailwindcss'
import { loadConfig } from 'tailwindcss-config'
import { getConfig } from './config'
import { postcssPlugin } from './constants'
import { regExpTest, removeFileExtension } from './utils'
import { getDepFiles } from './wxml'

const configCache = new Map<string, Promise<Config | undefined>>()

async function loadTailwindConfigOnce(cwd: string, configPath: string) {
  const cacheKey = `${cwd}::${configPath}`
  let cached = configCache.get(cacheKey)
  if (!cached) {
    cached = loadConfig({
      cwd,
      config: configPath,
    }).then(result => result?.config as Config | undefined)
    configCache.set(cacheKey, cached)
  }
  return cached
}

function cloneConfig(config: Config | undefined): Config | undefined {
  if (!config) {
    return config
  }

  return {
    ...config,
  }
}

async function resolveTailwindConfig(config: Options['config'], cwd: string, sourceInput: Input | undefined) {
  if (!config) {
    return undefined
  }

  const resolved = typeof config === 'function' ? config(sourceInput) : config

  if (typeof resolved === 'string') {
    const loaded = await loadTailwindConfigOnce(cwd, resolved)
    return cloneConfig(loaded)
  }

  return cloneConfig(resolved as Config)
}

// function isObject(obj: any): obj is object {
//   return typeof obj === 'object' && obj !== null
// }
export type {
  Options,
}

const creator: PluginCreator<Partial<Options>> = (options) => {
  const { config, filter, directiveParams, cwd, extensions, insertAfterAtRulesNames, insertAfterComments } = getConfig(options)

  const extensionsGlob = `{${extensions.join(',')}}`

  return {
    postcssPlugin,
    plugins: [
      {
        postcssPlugin: `${postcssPlugin}:post`,
        async Once(root, helpers) {
          const sourceInput = root.source?.input
          if (filter(sourceInput)) {
            const nodes = root.nodes ?? []
            let atruleAnchorIndex = -1
            let commentAnchorIndex = -1
            const directivePresence = new Set<string>()

            nodes.forEach((node, index) => {
              if (node.type === 'atrule') {
                const nameMatches = insertAfterAtRulesNames.includes(node.name)
                if (nameMatches) {
                  atruleAnchorIndex = Math.max(atruleAnchorIndex, index)
                }
                if (typeof node.params === 'string') {
                  directivePresence.add(node.params)
                }
              }
              else if (node.type === 'comment' && regExpTest(insertAfterComments, node.text)) {
                commentAnchorIndex = Math.max(commentAnchorIndex, index)
              }
            })

            const anchorIndex = Math.max(atruleAnchorIndex, commentAnchorIndex)
            const anchorNode = anchorIndex >= 0 ? nodes[anchorIndex] : undefined
            let lastInserted = anchorNode

            for (const params of directiveParams) {
              if (!directivePresence.has(params)) {
                const atRule = helpers.atRule({ name: 'tailwind', params })
                if (lastInserted) {
                  root.insertAfter(lastInserted, atRule)
                }
                else {
                  root.prepend(atRule)
                }
                lastInserted = atRule
                directivePresence.add(params)
              }
            }

            if (sourceInput?.file) {
              const tailwindcssConfig = (await resolveTailwindConfig(config, cwd, sourceInput)) ?? { content: [] }

              const basename = removeFileExtension(sourceInput.file)
              const contentEntries = [`${basename}.${extensionsGlob}`]

              const deps = await getDepFiles(`${basename}.wxml`)
              for (const dep of deps) {
                contentEntries.push(`${removeFileExtension(dep)}.${extensionsGlob}`)
              }

              const uniqueEntries = Array.from(new Set(contentEntries))
              const { content: existingContent } = tailwindcssConfig as { content?: Config['content'] }

              if (!existingContent || Array.isArray(existingContent)) {
                tailwindcssConfig.content = uniqueEntries
              }
              else if (typeof existingContent === 'object' && existingContent !== null) {
                const normalizedContent = { ...existingContent }
                const currentFiles = normalizedContent.files
                const normalizedFiles = Array.isArray(currentFiles)
                  ? currentFiles
                  : typeof currentFiles === 'string'
                    ? [currentFiles]
                    : []
                normalizedContent.files = Array.from(new Set([...normalizedFiles, ...uniqueEntries]))
                tailwindcssConfig.content = normalizedContent
              }
              else {
                tailwindcssConfig.content = uniqueEntries
              }

              await postcss([
                tailwindcss(tailwindcssConfig),
              ]).process(root, {
                from: sourceInput.file,
              }).async()
            }
          }
        },
      },
    ],
  }
}
creator.postcss = true

export default creator
