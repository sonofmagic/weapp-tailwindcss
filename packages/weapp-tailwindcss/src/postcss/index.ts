import postcss from 'postcss'
import type { IStyleHandlerOptions } from '../types'
import { defuOverrideArray } from '../utils'
import { getPlugins } from './plugins'

export function styleHandler(rawSource: string, options: IStyleHandlerOptions) {
  return postcss(getPlugins(options))
    .process(
      rawSource,
      options.postcssOptions?.options ?? {
        from: undefined,
      },
    )
    .async()
}

export function createStyleHandler(options: Partial<IStyleHandlerOptions>) {
  return (rawSource: string, opt?: Partial<IStyleHandlerOptions>) => {
    return styleHandler(
      rawSource,
      defuOverrideArray<IStyleHandlerOptions, Partial<IStyleHandlerOptions>[]>(opt as IStyleHandlerOptions, options),
    )
  }
}
