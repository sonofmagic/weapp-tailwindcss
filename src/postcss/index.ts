import postcss from 'postcss'
import { getPlugins } from './plugins'
import type { IStyleHandlerOptions } from '@/types'
import { defuOverrideArray } from '@/utils'

export async function styleHandler(rawSource: string, options: IStyleHandlerOptions) {
  return (await postcss(getPlugins(options)).process(rawSource).async()).css
}

export function createStyleHandler(options: Partial<IStyleHandlerOptions>) {
  return (rawSource: string, opt?: Partial<IStyleHandlerOptions>) => {
    return styleHandler(rawSource, defuOverrideArray<IStyleHandlerOptions, Partial<IStyleHandlerOptions>[]>(opt as IStyleHandlerOptions, options))
  }
}
