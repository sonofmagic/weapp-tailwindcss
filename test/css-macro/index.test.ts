// import plugin from 'tailwindcss/plugin'
import postcss from 'postcss'
// import twPlugin from '../../dist/css-macro'
import twPlugin from '@/css-macro'
import { getCss } from '#test/helpers/getTwCss'
import postcssPlugin from '@/css-macro/postcss'

// not screen and (weapp-tw-platform:MP-WEIXIN)
// not screen and (weapp-tw-platform:uniVersion > 3.9)
// not screen and (weapp-tw-platform:H5 || MP-WEIXIN)
describe('css-macro tailwindcss plugin', () => {
  it('dynamic case 0', async () => {
    const { css } = await getCss('ifdef-[MP-WEIXIN]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic case 1', async () => {
    const { css } = await getCss('ifndef-[MP-WEIXIN]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic case 2', async () => {
    const { css } = await getCss('ifndef-[uniVersion>3.9]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic case 3', async () => {
    const { css } = await getCss('ifndef-[uniVersion_>_3.9]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic case 4', async () => {
    const { css } = await getCss('ifndef-[uni\\_version_>_3.9]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic case 5', async () => {
    const { css } = await getCss('ifndef-[H5||MP-WEIXIN]:bg-blue-500', {
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('dynamic apply case 6', async () => {
    const { css } = await getCss('', {
      css: `.apply-test {
        @apply ifdef-[H5||MP-WEIXIN]:bg-blue-400 ifndef-[H5||MP-WEIXIN]:bg-red-400;
      }`,
      twConfig: {
        plugins: [twPlugin]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('static case 0', async () => {
    const { css } = await getCss('wx:bg-blue-500', {
      twConfig: {
        plugins: [
          twPlugin({
            variantsMap: {
              wx: 'MP-WEIXIN'
            }
          })
        ]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('static case 1', async () => {
    const { css } = await getCss('-wx:bg-blue-500', {
      twConfig: {
        plugins: [
          twPlugin({
            variantsMap: {
              '-wx': {
                value: 'MP-WEIXIN',
                negative: true
              }
            }
          })
        ]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('static case 2', async () => {
    const { css } = await getCss('hv:bg-blue-500', {
      twConfig: {
        plugins: [
          twPlugin({
            variantsMap: {
              hv: {
                value: 'uniVersion > 3.9'
              }
            }
          })
        ]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })

  it('static case 3', async () => {
    const { css } = await getCss('mv:bg-blue-500', {
      twConfig: {
        plugins: [
          twPlugin({
            variantsMap: {
              mv: {
                value: 'H5 || MP-WEIXIN'
              }
            }
          })
        ]
      }
    })
    expect(css).toMatchSnapshot('tw')
    const { css: cssOutput } = await postcss(postcssPlugin).process(css, {
      from: undefined
    })
    expect(cssOutput).toMatchSnapshot('postcss')
  })
})
