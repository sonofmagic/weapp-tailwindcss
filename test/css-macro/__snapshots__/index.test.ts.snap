// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css-macro tailwindcss plugin dynamic apply case 6: postcss 1`] = `
"
/* #ifdef H5||MP-WEIXIN */
    .apply-test {
        --tw-bg-opacity: 1;
        background-color: rgb(96 165 250 / var(--tw-bg-opacity))
    }
/* #endif */
/* #ifndef H5||MP-WEIXIN */
.apply-test {
        --tw-bg-opacity: 1;
        background-color: rgb(248 113 113 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic apply case 6: tw 1`] = `
"@media (weapp-tw-platform:"H5||MP-WEIXIN") {
    .apply-test {
        --tw-bg-opacity: 1;
        background-color: rgb(96 165 250 / var(--tw-bg-opacity))
    }
}
@media not screen and (weapp-tw-platform:"H5||MP-WEIXIN") {
    .apply-test {
        --tw-bg-opacity: 1;
        background-color: rgb(248 113 113 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 0: postcss 1`] = `
"
/* #ifdef MP-WEIXIN */
    .ifdef-\\[MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 0: tw 1`] = `
"@media (weapp-tw-platform:"MP-WEIXIN") {
    .ifdef-\\[MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 1: postcss 1`] = `
"
/* #ifndef MP-WEIXIN */
    .ifndef-\\[MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 1: tw 1`] = `
"@media not screen and (weapp-tw-platform:"MP-WEIXIN") {
    .ifndef-\\[MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 2: postcss 1`] = `
"
/* #ifndef uniVersion>3.9 */
    .ifndef-\\[uniVersion\\>3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 2: tw 1`] = `
"@media not screen and (weapp-tw-platform:"uniVersion>3.9") {
    .ifndef-\\[uniVersion\\>3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 3: postcss 1`] = `
"
/* #ifndef uniVersion > 3.9 */
    .ifndef-\\[uniVersion_\\>_3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 3: tw 1`] = `
"@media not screen and (weapp-tw-platform:"uniVersion_>_3.9") {
    .ifndef-\\[uniVersion_\\>_3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 4: postcss 1`] = `
"
/* #ifndef uni\\_version > 3.9 */
    .ifndef-\\[uni\\\\_version_\\>_3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 4: tw 1`] = `
"@media not screen and (weapp-tw-platform:"uni\\_version_>_3.9") {
    .ifndef-\\[uni\\\\_version_\\>_3\\.9\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin dynamic case 5: postcss 1`] = `
"
/* #ifndef H5||MP-WEIXIN */
    .ifndef-\\[H5\\|\\|MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin dynamic case 5: tw 1`] = `
"@media not screen and (weapp-tw-platform:"H5||MP-WEIXIN") {
    .ifndef-\\[H5\\|\\|MP-WEIXIN\\]\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin fix comment eol case 0: postcss 1`] = `
"/* #ifndef MP-WEIXIN */
   .-wxcbg-red-500 {
     --tw-bg-opacity: 1;
     background-color: rgb(239 68 68 / var(--tw-bg-opacity));
   }
   /* #endif */
   "
`;

exports[`css-macro tailwindcss plugin postcss expand case 0: postcss 1`] = `
"
/* #ifndef MP-WEIXIN */
      .-wxcbg-red-500 {
          --tw-bg-opacity: 1;
          background-color: rgb(239 68 68 / var(--tw-bg-opacity));
      }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin postcss expand case 1: postcss 1`] = `
"
/* #ifndef MP-WEIXIN */
        .ifndef-_MP-WEIXIN_cbg-red-500 {
            --tw-bg-opacity: 1;
            background-color: rgb(239 68 68 / var(--tw-bg-opacity));
        }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin static case 0: postcss 1`] = `
"
/* #ifdef MP-WEIXIN */
    .wx\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin static case 0: tw 1`] = `
"@media (weapp-tw-platform:"MP-WEIXIN") {
    .wx\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin static case 1: postcss 1`] = `
"
/* #ifndef MP-WEIXIN */
    .-wx\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin static case 1: tw 1`] = `
"@media not screen and (weapp-tw-platform:"MP-WEIXIN") {
    .-wx\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin static case 2: postcss 1`] = `
"
/* #ifdef uniVersion > 3.9 */
    .hv\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin static case 2: tw 1`] = `
"@media (weapp-tw-platform:"uniVersion > 3.9") {
    .hv\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;

exports[`css-macro tailwindcss plugin static case 3: postcss 1`] = `
"
/* #ifdef H5 || MP-WEIXIN */
    .mv\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
/* #endif */"
`;

exports[`css-macro tailwindcss plugin static case 3: tw 1`] = `
"@media (weapp-tw-platform:"H5 || MP-WEIXIN") {
    .mv\\:bg-blue-500 {
        --tw-bg-opacity: 1;
        background-color: rgb(59 130 246 / var(--tw-bg-opacity))
    }
}"
`;
